l10n_pt_account/guia.py

```diff
@@ -0,0 +1,663 @@
+# -*- coding: utf-8 -*-
+# Copyright 2010-2016 ThinkOpen Solutions
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)./
+
+from openerp import api, fields, models, _
+from openerp.exceptions import Warning
+
+from openerp import tools
+import openerp.addons.decimal_precision as dp
+
+from datetime import datetime, date, timedelta
+
+HOLIDAYS = set([
+    (1, 1),  # Ano Novo
+    (4, 25),  # Dia da Liberdade
+    (5, 1),  # Dia do Trabalhador
+    (6, 10),  # Dia de Portugal
+    (8, 15),  # Assunção de Nossa Senhora
+    (12, 8),  # Imaculada Conceição
+    (12, 25),  # Natal
+])
+
+
+def is_holiday(day):
+    if (day.month, day.day) in HOLIDAYS:
+        return True
+    # Calculate Easter
+    # from
+    # http://www.forma-te.com/mediateca/download-
+    #  document/5855-como-calcular-os-feriados-moveis.html
+    X, Y = 24, 5
+    a = day.year % 19
+    b = day.year % 4
+    c = day.year % 7
+    d = (19 * a + X) % 30
+    e = (2 * b + 4 * c + 6 * d + Y) % 7
+    easter = date(day.year, 4, d + e - 9) if d + \
+        e > 9 else date(day.year, 3, d + e + 22)
+    sexta_feira_santa = easter - timedelta(days=2)
+    return d in (easter, sexta_feira_santa)
```

Comment:
As a future improvement, holidays could be moved to their own module, and local holidays could then be added.